lib/quickbooks/model/base_model.rb

```diff
@@ -12,6 +12,48 @@ def initialize(attributes={})
         attributes.each {|key, value| public_send("#{key}=", value) }
       end
 
+      def self.all query = nil, options = {}
+        model_name           = self.name.split('::').last
+        query                = "SELECT * FROM #{model_name}" if query.blank?
+        service              = eval("Quickbooks::Service::#{model_name}").new
+        service.access_token = OAuth::AccessToken.new(
+          $qb_oauth_consumer, options[:token] || Thread.current[:token],
+          options[:secret] || Thread.current[:secret]
+        )
+        service.company_id   = options[:realm_id] || Thread.current[:realm_id]
+        service.query(query).entries
+      end
+
+      def self.where options = {}
+        model_name = self.name.split('::').last
+        query      = options.present? ? "SELECT * FROM #{model_name} WHERE " : "SELECT * FROM #{model_name}"
+        if options.is_a?(String)
+          conditions = options
+        else
+          conditions = options.collect do |key, value|
+            if value.is_a?(Array)
+              "#{key} IN ('#{value.join("','")}')"
```

Comment:
`value` should leverage Quickbooks::Util::QueryBuilder e.g.
https://github.com/minimul/quickbooks-ruby-base/blob/master/lib/quickbooks/base/finders.rb